core/tee/fs_htree.c

```diff
@@ -707,6 +712,13 @@ static TEE_Result htree_sync_node_to_storage(struct traverse_arg *targ,
 	TEE_Result res;
 	uint8_t vers;
 
+	/*
+	 * The node can be dirty while the block isn't updated due to
+	 * updated children, but if block is updated the node has to be
+	 * dirty.
+	 */
+	assert(node->dirty >= node->block_updated);
```

Comment:
The standard says
> An object declared as type _Bool is large enough to store the values 0 and 1.

So I can't see how this could be invalid (if it's a style issue or not is another question though). I think we're even doing bitwise or with bool at some other place in the code.